open_flamingo/eval/evaluate.py

```diff
@@ -473,351 +276,125 @@ def main():
-                f"{args.cached_demonstration_features}/textvqa.pkl", map_location="cpu"
-            )
-        else:
-            cached_features = None
-
-        for shot in args.shots:
-            scores = []
-            for seed, trial in zip(args.trial_seeds, range(args.num_trials)):
-                textvqa_score = evaluate_vqa(
-                    args=args,
-                    eval_model=eval_model,
-                    num_shots=shot,
-                    seed=seed,
-                    dataset_name="textvqa",
-                    max_new_tokens=10,
-                    cached_features=cached_features,
-                )
                 if args.rank == 0:
-                    print(f"Shots {shot} Trial {trial} TextVQA score: {textvqa_score}")
-                    scores.append(textvqa_score)
-
-            if args.rank == 0:
-                print(f"Shots {shot} Mean TextVQA score: {np.nanmean(scores)}")
-                results["textvqa"].append(
-                    {
-                        "shots": shot,
-                        "trials": scores,
-                        "mean": np.nanmean(scores),
-                        "stddev": np.nanstd(scores),
-                    }
-                )
+                    print(f"Shots {shot} Mean CIDEr score: {np.nanmean(scores)}")
+                    results[task].append(
+                        {
+                            "shots": shot,
+                            "trials": scores,
+                            "mean": np.nanmean(scores),
+                            "stddev": np.nanstd(scores),
+                        }
+                    )
     
-    if args.eval_gqa:
-        print("Evaluating on GQA...")
-
-        #load cached demonstration features on GQA
-        if args.cached_demonstration_features is not None:
-            cached_features = torch.load(
-                f"{args.cached_demonstration_features}/imagenet.pkl", map_location="cpu"
-            )
-        else:
-            cached_features = None
-        
-        for shot in args.shots:
-            scores = []
-            for seed, trial in zip(args.trial_seeds, range(args.num_trials)):
-                gqa_score = evaluate_vqa(
-                    args=args,
-                    eval_model=eval_model,
-                    num_shots=shot,
-                    seed=seed,
-                    dataset_name="gqa",
-                    max_new_tokens=10,
-                    cached_features=cached_features,
+    for vqa_task in ["okvqa", "vqav2", "vizwiz", "textvqa", "gqa"]:
+        if var_args[f"eval_{vqa_task}"]:
+            print(f"Evaluating on {vqa_task}...")
+
+            # load cached demonstration features for RICES
+            if args.cached_demonstration_features is not None:
+                cached_features = torch.load(
+                    f"{args.cached_demonstration_features}/{'ok_vqa' if vqa_task=='okvqa' else vqa_task}.pkl", map_location="cpu"
```

Comment:
can we unify the way we refer to okvqa? Like let's just stick to okvqa as a single word.